File: Control/TeslaCommunication/Packets/AbstractInPacket.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Runtime.InteropServices;
using System.Text;

namespace TeslaCommunication.Packets
{
    public abstract class AbstractInPacket
    {

        public int Command { get; set; }

        public int BodySize { get; set; }
        

        public abstract void ApplyBody(Byte[] buf, int offset, int size);
        public abstract void Process();
    }
}

File: Протокол тело 2 байта размер/Packets/AbstractInPacket.java
package ru.track_it.motohelper.Packets;

public abstract class AbstractInPacket {

    public int Command;

    public int BodySize;

    //called in packet former thread (should be fast)
    public abstract void ApplyBody(byte[] buf, int offset, int size);

    //действие по умолчанию для пакета (вызывается в другом потоке и
    //может быть длительным)
    public void DefaultProcess()
    {

    }

}
